src/auto_archiver/archivers/generic_archiver/generic_archiver.py

```diff
@@ -0,0 +1,297 @@
+
+        # then add the platform specific additional metadata
+        for key, mapping in self.video_data_metadata_mapping(extractor_key, video_data).items():
+            if isinstance(mapping, str):
+                result.set(key, eval(f"video_data{mapping}"))
+            elif callable(mapping):
+                result.set(key, mapping(video_data))
+        result.set_url(url)
+
+        # extract comments if enabled
+        if self.comments:
+            result.set("comments", [{
+                "text": c["text"],
+                "author": c["author"], 
+                "timestamp": datetime.datetime.fromtimestamp(c.get("timestamp"), tz = datetime.timezone.utc)
+            } for c in video_data.get("comments", [])])
+
+        # then add the common metadata
+        if timestamp := video_data.pop("timestamp", None):
+            timestamp = datetime.datetime.fromtimestamp(timestamp, tz = datetime.timezone.utc).isoformat()
+            result.set_timestamp(timestamp)
+        if upload_date := video_data.pop("upload_date", None):
+            upload_date = datetime.datetime.strptime(upload_date, '%Y%m%d').replace(tzinfo=datetime.timezone.utc)
+            result.set("upload_date", upload_date)
+        
+        # then clean away any keys we don't want
+        for clean_key in self.keys_to_clean(extractor_key, video_data):
+            video_data.pop(clean_key, None)
+        
+        # then add the rest of the video data
+        for k, v in video_data.items():
+            if v:
+                result.set(k, v)
+
+        return result
+
+    def video_data_metadata_mapping(self, extractor_key: str, video_data: dict) -> dict:
+        """
+        Returns a key->value mapping to map from the yt-dlp produced 'video_data' to the Metadata object.
+        Can be either a string for direct mapping, or a function, or a lambda.
+        """
+        return {}
+    
+    def suitable_extractors(self, url: str) -> list[str]:
+        """
+        Returns a list of valid extractors for the given URL"""
+        for info_extractor in yt_dlp.YoutubeDL()._ies.values():
+            if info_extractor.suitable(url) and info_extractor.working():
+                yield info_extractor
+        
+    def suitable(self, url: str) -> bool:
+        """
+        Checks for valid URLs out of all ytdlp extractors.
+        Returns False for the GenericIE, which as labelled by yt-dlp: 'Generic downloader that works on some sites'
+        """
+        return any(self.suitable_extractors(url))
+
+    def create_metadata_for_post(self, info_extractor: InfoExtractor, video_data: dict, url: str) -> Metadata:
+        """
+        Standardizes the output of the ytdlp InfoExtractor to a common format
+        """
+        if info_extractor.ie_key() == 'Bluesky':
+            return bluesky.create_metadata(video_data, self, url)
+        if info_extractor.ie_key() == 'Twitter':
+            return twitter.create_metadata(video_data, self, url)
```

Comment:
tiktok missing here?

Reply:
This is a good question. So actually it's not needed for tiktok, since all tiktok pages have valid videos, and the 'standard' video_data is returned by the ytdl tiktok extractor. So no further processing is required.

But you've made me realise I can abstract out and make the code clearer. I will do that